modules/internal/DefaultRectangular.chpl

```diff
@@ -1400,502 +1401,244 @@ module DefaultRectangular {
-    var Blo: rank*dom.idxType;
-    for param i in 1..rank do
-      Blo(i) = Bdims(i).first;
-    
-    if debugDefaultDistBulkTransfer then
-      writeln("\tlocal get() from ", B.locale.id);
-    
-    var dstWholeDim = isWholeDim(A),
-        srcWholeDim = isWholeDim(B);
-    var stridelevels:int(32);
-    
-    /* If the stridelevels in source and destination arrays are different, we take the larger*/
-    stridelevels=max(A.computeBulkStrideLevels(dstWholeDim),B.computeBulkStrideLevels(srcWholeDim));
-    if debugDefaultDistBulkTransfer then 
-      writeln("In DefaultRectangularArr.doiBulkTransferStride, stridelevels: ",stridelevels);
-    
-    //These variables should be actually of size stridelevels+1, but stridelevels is not param...
-    
-    var srcCount, dstCount:[1..rank+1] size_t;
-    
-    // Covering the case in which stridelevels has to be incremented after
-    // unifying srcCount and dstCount into a single count array. To illustrate the problem:
-    
-    // config const n = 10 
-    // var A: [1..n,1..n] real(64);
-    // var B: [1..n,1..n] real(64);
-    
-    // A[1..10,1..5] = B[1..10, 1..10 by 2]
-    
-    // Computed variables for chpl_comm_get_strd/puts are:
-    //  stridelevels =1
-    //  srcCount = (1,50) //In B you read 1 element 50 times
-    //  (srcStride=(2) = distance between 1 element and the next one)
-    //  dstCount = (5,10) //In A you write a chunk of 5 elements 10
-    //  times (dstStride=(10) distance between 1 chunk and the next one)
-    
-    
-    // Since GASNet strided put/get only have a count array, it is
-    // necessary to choose the count array that follows the smaller size of
-    // the chunks either at the source or destination array. That way the
-    // right unified cnt=(1,5,10), which forces to increment stridelevels to
-    // 2 and and now srcStride=(2,10) and dstStride=(1,10).
-    var dstAux:bool = false;
-    var srcAux:bool = false;
-    if rank > 1 then
-      if (A.dom.dsiDim(rank).stride>1 && B.dom.dsiDim(rank).stride==1)
-      {
-        if stridelevels < rank then stridelevels+=1;
-        dstAux = true;
-      }
-      else if (B.dom.dsiDim(rank).stride>1 && A.dom.dsiDim(rank).stride==1)
-      {
-        if stridelevels < rank then stridelevels+=1;
-        srcAux = true;
+    var BFirst : rank*idxType;
+    for i in 1..rank do BFirst(i) = Bdims(i).first;
+
+    if debugDefaultDistBulkTransfer {
+      writeln("In DefaultRectangularArr.doiBulkTransferStride");
+      writeln("Dest = ", Adims);
+      writeln("Src  = ", Bdims);
+    }
+
+    // The number of values needed to express the strided region.
+    var stridelevels = 0;
+
+    //
+    // `count[1]` is the number of elements we can bulk-copy at a time. Each
+    // number after that represents the number of times we need to stride at
+    // each level. It will ultimately be an array of size `stridelevels+1`.
+    //
+    var countDom = {1..rank+1};
+    var count : [countDom] int;
```

Comment:
I intentionally did a serial assignment to avoid the overhead of task creation for such a small array. I can document that rationale though.